### PR TITLE
Add Schema and BitmapSchema

### PR DESCRIPTION
**Pull request by mfucci**
This adds the core definition of a generic schema, with one implementation to encode / decode Bitmaps.

### CODE REVIEW

File: src/schema/BitmapSchema.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+/**
+ * @file Define schema for a bitmap.
+ * @copyright Project CHIP Authors 2022
+ * @license Apache-2.0
+ */
```
Comment (mfucci) [inline src/schema/BitmapSchema.ts:5]: What do you think of using those short license / copyrights declarations?
They follow the JsDoc 3 from here: https://jsdoc.app/

Comment (turon) [reply src/schema/BitmapSchema.ts:5]: This is the compliant license style:
```
/**
 * @license
 * Copyright 2000 Example, Inc
 * SPDX-License-Identifier: Apache-2.0
 */
```

Comment (mfucci) [reply src/schema/BitmapSchema.ts:5]: Compliant with Matter license declaration?

Per JsDoc 3 doc, the SPDX Id can be used directly after @license:
https://jsdoc.app/tags-license.html

For the @copyright tag, it is supposed to use in combination with a @file tag:
https://jsdoc.app/tags-copyright.html
Maybe should be @copyright 2022 Project CHIP Authors to follow the recommendation from inside Apache 2.0 license
https://github.com/project-chip/matter.js/blob/main/LICENSE#L189

I don't mind using your suggestion above or the full license declaration, I just thought the short version was neat...

Comment (turon) [reply src/schema/BitmapSchema.ts:5]: Yes, I like the proposed format too, but for now it is most expedient to stay with the format that has been pre-approved by legal.

Comment (mfucci) [reply src/schema/BitmapSchema.ts:5]: Ok, I will update that when I get a break in my day time job! (In 3h...)

Comment (mfucci) [reply src/schema/BitmapSchema.ts:5]: Updated

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+/**
+ * @license
+ * Copyright 2000 Example, Inc
```
Comment (turon) [inline src/index.ts:3]:
```suggestion
 * Copyright 2022 Project CHIP Authors
```

File: test/schema/BitmapSchemaTest.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
+/**
+ * @license
+ * Copyright 2000 Example, Inc
```
Comment (turon) [inline test/schema/BitmapSchemaTest.ts:3]:
```suggestion
 * Copyright 2022 Project CHIP Authors
```

File: src/schema/BitmapSchema.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+/** Specifies the position of the bit. */
+export interface BitPosition { position: number }
+export const BitPosition = (position: number) => ({ position });
+
+type BitSchema = {[key: string]: BitPosition};
+type TypeFromBitSchema<T extends BitSchema> = {[K in keyof T]: boolean};
+
+/** Declares a bitmap schema by indicating the bit position and their names. */
+export const BitmapSchema = (bitSchemas: BitSchema) => new class<T extends BitSchema> extends Schema<TypeFromBitSchema<T>, number>{
```
Comment (turon) [inline src/schema/BitmapSchema.ts:17]: Would it be possible to simplify this definition to use simple `export class<T> ...` syntax?

**Review by turon — APPROVED** [review]

File: src/schema/BitmapSchema.ts (hunk 4 above)
Comment (mfucci) [reply src/schema/BitmapSchema.ts:17]: Unfortunately not and this declaration was actually buggy.
Fixed now but even more complex...

**Pull request merged**